src/v2.rs

```diff
@@ -24,20 +29,143 @@ use crate::Module;
 use crate::ModuleInner;
 pub use crate::ModuleKind;
 
-pub(crate) const ESZIP_V2_MAGIC: &[u8; 8] = b"ESZIP_V2";
+const ESZIP_V2_MAGIC: &[u8; 8] = b"ESZIP_V2";
+const ESZIP_V3_MAGIC: &[u8; 8] = b"ESZIP_V3";
```

Comment:
@lucacasonato can't do v2.1 here because this is already 8 bytes. Maybe what's done here is ok? (I guess the API is versioned separately from the file format)